### PR TITLE
feat(executor): Allow to configure the interval of resource state checking

### PR DESCRIPTION
**Pull request by terrytangyuan**
See some context in https://github.com/argoproj/argo/issues/4467. This is a workaround that could reduce the burden of the cluster.

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/executor/resource.go [hunk 1]
```diff
@@ -165,8 +169,8 @@ func (we *WorkflowExecutor) WaitResource(resourceNamespace string, resourceName
 	}
 
 	// Start the condition result reader using PollImmediateInfinite
-	// Poll intervall of 5 seconds serves as a backoff intervall in case of immediate result reader failure
-	err := wait.PollImmediateInfinite(time.Second*5,
+	// Poll interval serves as a backoff interval in case of immediate result reader failure
+	err := wait.PollImmediateInfinite(resourceStateCheckInterval,
```
Comment (jessesuen) [inline workflow/executor/resource.go:173]: This change is not really necessary and will have no affect on performance. 

`checkResourceState` is actually a long, blocking watch on the resource and not a not a get. So the only time the interval comes into play is when the watch fails, which should be rare.

**Review by jessesuen — CHANGES_REQUESTED** [review]
I don't think this change is necessary.

File: workflow/executor/resource.go (hunk 1 above)
Comment (terrytangyuan) [reply workflow/executor/resource.go:173]: Got it. I previously missed the `-w` part and thought it was a get. So looks like there should be no performance issue here and https://github.com/argoproj/argo/issues/4467 is no longer necessary. Am I right?

Comment (jessesuen) [reply workflow/executor/resource.go:173]: Correct. I think we should close this.

**Pull request closed without merging**